kernel/src/hil/framebuffer.rs

```diff
@@ -0,0 +1,164 @@
+    #[repr(usize)]
+    #[allow(non_camel_case_types)]
+    pub enum ScreenPixelFormat {
+        /// Pixels encoded as 1-bit, used for monochromatic displays
+        Mono = 0,
+        /// Pixels encoded as 2-bit red channel, 3-bit green channel, 3-bit blue channel.
+        RGB_233 = 1,
+        /// Pixels encoded as 5-bit red channel, 6-bit green channel, 5-bit blue channel.
+        RGB_565 = 2,
+        /// Pixels encoded as 8-bit red channel, 8-bit green channel, 8-bit blue channel.
+        RGB_888 = 3,
+        /// Pixels encoded as 8-bit alpha channel, 8-bit red channel, 8-bit green channel, 8-bit blue channel.
+        ARGB_8888 = 4,
+        // other pixel formats may be defined.
+    }
+}
+
+impl ScreenPixelFormat {
+    pub fn get_bits_per_pixel(&self) -> usize {
+        match self {
+            Self::Mono => 1,
+            Self::RGB_233 => 8,
+            Self::RGB_565 => 16,
+            Self::RGB_888 => 24,
+            Self::ARGB_8888 => 32,
+        }
+    }
+}
+
+pub trait ScreenSetup {
+    fn set_client(&self, client: Option<&'static dyn ScreenSetupClient>);
+
+    /// Sets the screen resolution (in pixels). Returns ENOSUPPORT if the resolution is
+    /// not supported. The function should return SUCCESS if the request is registered
+    /// and will be sent to the screen.
+    /// Upon SUCCESS, the caller has to wait for the `command_complete` callback function
+    /// that will return the actual ReturnCode after setting the resolution.
+    fn set_resolution(&self, resolution: (usize, usize)) -> ReturnCode;
+
+    /// Sets the pixel format. Returns ENOSUPPORT if the pixel format is
+    /// not supported. The function should return SUCCESS if the request is registered
+    /// and will be sent to the screen.
+    /// Upon SUCCESS, the caller has to wait for the `command_complete` callback function
+    /// that will return the actual ReturnCode after setting the pixel format.
+    fn set_pixel_format(&self, depth: ScreenPixelFormat) -> ReturnCode;
+
+    /// Sets the rotation of the display.
+    /// The function should return SUCCESS if the request is registered
+    /// and will be sent to the screen.
+    /// Upon SUCCESS, the caller has to wait for the `command_complete` callback function
+    /// that will return the actual ReturnCode after setting the rotation.
+    ///
+    /// note this can swap the width with height.
+    fn set_rotation(&self, rotation: ScreenRotation) -> ReturnCode;
+
+    /// Returns the number of the resolutions supported.
+    /// should return at least one (the current resolution)
+    /// This function is synchronous as the driver should know this value without
+    /// requesting it from the screen (most screens do not support such a request,
+    /// resolutions are described in the data sheet).
+    ///
+    /// If the screen supports such a feature, the driver should request this information
+    /// from the screen upfront.
+    fn get_num_supported_resolutions(&self) -> usize;
```

Comment:
I'm not sure I understand how this would be used. I imagine it would be easy to implement, but what can clients expect to do with the number of supported resolutions?

Reply:
Some screens like ST7735 seem to support several resolution, I thought the user might want to select one.